lib/TPP/Dialect/Tpp/TppOps.cpp

```diff
@@ -321,30 +357,54 @@ void GemmOp::getEffects(
 // BrgemmOp
 //===----------------------------------------------------------------------===//
 
-static bool verifyBRGemmShape(ShapedType shapedA, ShapedType shapedB,
-                              ShapedType shapedC) {
-  return !(shapedB.getRank() != 3 || shapedC.getRank() != 2 ||
-           shapedA.getRank() != 3);
-}
+template <typename OpTy>
+static LogicalResult verifyBgemmOperands(OpTy operation) {
```

Comment:
Can you common `gemm` and `brgemm` verification, at least for a subset of the checks? They look very similar.